### PR TITLE
Fixes timezone issues and routes with legs with type=WALK

### PR DESCRIPTION
**Pull request by kraenhansen**
This fixes two issues:

1. Timezone - the communication with "Rejseplanen" didn't take daylight savings into account, returning routes that we too far into the future. This is fixed by explicitly stating the current time in Copenhagen when requesting trips from Rejseplanen and specifying the timezone when parsing results from their API.
2. Rejseplanen returned routes with legs that had type = "WALK" and no JourneyDetailRef field. This made the request fail. Now we parse walking as we do biking.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,7 +9,7 @@ gem 'bcrypt', require: 'bcrypt'
 gem 'devise', '~> 4.0.2'
 gem 'omniauth'
 gem 'omniauth-facebook'
-gem 'devise-token_authenticatable', '~> 0.5.0.beta'
+gem 'devise-token_authenticatable', '~> 0.5.1'
```
Comment (emiltin) [inline Gemfile]: is this necessary to fix the rejseplanen api problem?

**Comment by emiltin** [comment]
could you rebase on latest develop?

File: app/controllers/api/v1/journey_controller.rb [hunk 2]
```diff
@@ -1,11 +1,12 @@
 class Api::V1::JourneyController < Api::V1::BaseController
   skip_before_filter :check_auth_token!
+  skip_before_filter :verify_authenticity_token
 
   rescue_from TravelPlanner::Error, with: :travel_planner_message
   rescue_from StandardError, with: :standard_message
 
   def create
-    remote_outdated_journeys
+    remove_outdated_journeys
```
Comment (emiltin) [inline app/controllers/api/v1/journey_controller.rb]: thanks for catching this

File: app/lib/travel_planner/journey.rb [hunk 3]
```diff
@@ -24,7 +24,12 @@ def format_legs(journey_data)
     journey = journey_data['Leg'].map do |leg_data|
       leg = TravelPlanner::Leg.new leg_data, @coords
 
-      formatted_leg = leg.type=='BIKE' ? format_bike(leg) : format_public(leg)
+      formatted_leg = case leg.type
+      when 'BIKE', 'WALK'
+        format_bike(leg)
```
Comment (emiltin) [inline app/lib/travel_planner/journey.rb]: what happens if you if:
you can't bike? (i assume you then push the bike)
you can't even push a bike?

File: app/lib/travel_planner/leg.rb [hunk 4]
```diff
@@ -94,7 +95,7 @@ def build_route_instructions
         '19',
         destination['name'],
         distance,
-        1,
+        1, # FIXME: This should be waypoints.size()-1 instead
```
Comment (emiltin) [inline app/lib/travel_planner/leg.rb]: what is the current behaviour without fixing this?

File: Gemfile (hunk 1 above)
Comment (kraenhansen) [inline Gemfile]: Yep - "bundle install" will fail when pushing to Heroku

File: app/lib/travel_planner/journey.rb (hunk 3 above)
Comment (kraenhansen) [inline app/lib/travel_planner/journey.rb]: That is for the route planner to decide - but now we are at least asking it, if the Rejseplanen has a leg where they think the user should walk

File: app/lib/travel_planner/leg.rb (hunk 4 above)
Comment (kraenhansen) [inline app/lib/travel_planner/leg.rb]: The apps currently compensate for the error by not using the value

**Comment by kraenhansen** [comment]
I've rebased current develop onto the branch

**Comment by emiltin** [comment]
@kraenhansen you wrote "rebased current develop onto the branch", but you need to rebase this branch on top of current develop, not the other way around, ie

git checkout fix/timezone+walk
git rebase origin/develop

maybe this is what you've already done?

**Comment by emiltin** [comment]
is the commit "Upgraded devise-token_authenticatable" related to this PR?

**Comment by kraenhansen** [comment]
As I commented above - bundle installing when Heroku is running bin/setup will fail as "devise-token_authenticatable" is no longer available for installation.

**Comment by kraenhansen** [comment]
Regarding the rebase - I think this is a matter of words. I did what the same as you wrote, so everything is pretty https://github.com/ibikecph/ibikecph-site/network

**Comment by emiltin** [comment]
super. gonna merge.

**Pull request merged**